src/core/reaction_methods/utils.cpp

```diff
@@ -66,4 +66,26 @@ calculate_factorial_expression(SingleReaction const &current_reaction,
   return factorial_expr;
 }
 
+double
+calculate_factorial_expression_cpH(SingleReaction const &current_reaction,
+                               std::map<int, int> const &old_particle_numbers) {
+  double factorial_expr = 1.0;
+  // factorial contribution of reactants
+  for (int i = 0; i < 1; i++) {
+    int nu_i = -1 * current_reaction.reactant_coefficients[i];
```

Comment:
The trick here is that i can only take the value 0. This assumes that A and HA are provided always in the first entry of the educt and product list!